pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -220,6 +220,13 @@ public PgConnection(HostSpec[] hostSpecs,
     // Now make the initial connection and set up local state
     this.queryExecutor = ConnectionFactory.openConnection(hostSpecs, user, database, info, logger);
 
+    // Reject connection to unsupported servers (8.1 and lower are not supported)
+    if (this.queryExecutor.getServerVersionNum() < 80200) {
```

Comment:
Could you use `ServerVersion` somehow?
https://github.com/pgjdbc/pgjdbc/blob/14e64be7dc43feaef210e82fcd9b6089e54e5e4f/pgjdbc/src/main/java/org/postgresql/core/ServerVersion.java#L17